doc/src/errorrecovery.md

```diff
@@ -299,6 +289,47 @@ Result: 2
 ```
 
 
+## Biasing repair sequences
+
+Depending on your language, some repair sequences are better than others. For
+example, sometimes `Insert` repairs are less welcome than `Delete` repairs:
+
+```ignore
+>>> 2 + + 3
+Parsing error at line 1 column 3. Repair sequences found:
+   1: Insert INT
+   2: Delete +
+Unable to evaluate expression.
+>>> 2 + + 3
+Parsing error at line 1 column 3. Repair sequences found:
+   1: Delete +
+   2: Insert INT
+Result: 5
+```
+
+Why does the same input sometimes produce a result and sometimes fail to produce
+a result? The problem is that `2 + + 3` has two repair sequences `Delete +` and
+`Insert Int`. As things stand, both are equally good, and so one is chosen
+non-deterministically. If `Insert Int` is chosen, we hit the `Err` case from
+earlier, and fail to produce a result; if the `Delete` case is chosen, we can
+produce a result.
+
+There is thus a simple mechanism to bias error recovery away from inserting
+lexemes that cause grammar actions problem. `%avoid_insert L` causes grmtools to
```

Comment:
`that cause the grammar actions problem` or `that cause grammar actions problems`.

Reply:
Hmm, this sentence is a bit awkward isn't it? Let me try rephrasing.